src/test/java/org/prebid/server/privacy/gdpr/TcfDefinerServiceTest.java

```diff
@@ -112,14 +114,93 @@ public void resultForVendorIdsShouldAllowAllWhenGdprIsDisabled() {
+                .enabledForRequestType(EnabledForRequestType.of(null, false, null, null))
+                .build();
+
+        // when
+        final Future<TcfResponse<String>> result = target.resultForBidderNames(
+                singleton("b"), null, null, null, null, accountGdprConfig, MetricName.amp, null);
+
+        // then
+        assertThat(result).succeededWith(
+                TcfResponse.of(false, singletonMap("b", PrivacyEnforcementAction.allowAll()), null));
+
+        verifyZeroInteractions(gdprService);
+        verifyZeroInteractions(tcf2Service);
+        verifyZeroInteractions(geoLocationService);
+        verifyZeroInteractions(metrics);
+    }
+
     @Test
     public void resultForBidderNamesShouldAllowAllWhenGdprIsDisabledByAccount() {
         // given
         final AccountGdprConfig accountGdprConfig = AccountGdprConfig.builder().enabled(false).build();
 
         // when
         final Future<TcfResponse<String>> result = target.resultForBidderNames(
-                singleton("b"), null, null, null, null, accountGdprConfig, null);
+                singleton("b"), null, null, null, null, accountGdprConfig, null, null);
+
+        // then
+        assertThat(result).succeededWith(
+                TcfResponse.of(false, singletonMap("b", PrivacyEnforcementAction.allowAll()), null));
+
+        verifyZeroInteractions(gdprService);
+        verifyZeroInteractions(tcf2Service);
+        verifyZeroInteractions(geoLocationService);
+        verifyZeroInteractions(metrics);
+    }
+
+    @Test
+    public void resultForBidderNamesShouldAllowAllWhenGdprIsNotDefinedForRequestTypeAndPBSGdprIsDisabled() {
+        // given
+        final GdprConfig gdprConfig = GdprConfig.builder()
+                .defaultValue("1")
+                .enabled(false)
+                .purposes(purposes)
+                .build();
+        final TcfDefinerService target = new TcfDefinerService(gdprConfig, singleton(EEA_COUNTRY), gdprService,
+                tcf2Service, geoLocationService, bidderCatalog, metrics);
+
+        final AccountGdprConfig accountGdprConfig = AccountGdprConfig.builder()
+                .enabledForRequestType(EnabledForRequestType.of(null, null, null, null))
+                .build();
+
+        // when
+        final Future<TcfResponse<String>> result = target.resultForBidderNames(
+                singleton("b"), null, null, null, null, accountGdprConfig, MetricName.amp, null);
+
+        // then
+        assertThat(result).succeededWith(
+                TcfResponse.of(false, singletonMap("b", PrivacyEnforcementAction.allowAll()), null));
+
+        verifyZeroInteractions(gdprService);
+        verifyZeroInteractions(tcf2Service);
+        verifyZeroInteractions(geoLocationService);
+        verifyZeroInteractions(metrics);
+    }
+
+    @Test
+    public void resultForBidderNamesShouldIgnoreUnknownRequestTypeAndDecideByPbsGdprDisabledProperty() {
```

Comment:
Don't we need to check account.enabled also ?

Reply:
No, we don't. Account.enabled only for usersync and setuid. If gdpr for request type is not defined only pbs config checked.